File: assiist_front_end/lib/core/repositories/account_repository.dart
import 'package:assiist_front_end/core/models/account_details.dart';

/// Abstract repository for managing account-level operations.
abstract class AccountRepository {
  /// Gets the account ID for the current user.
  ///
  /// Returns the account ID as a string.
  /// Throws an [ApiException] or its subclasses if the operation fails.
  Future<String> getAccountId();

  /// Fetches the account's details (business description and type).
  ///
  /// Returns [AccountDetailsResponse] containing the details.
  /// Returns null or throws an exception upon failure, depending on implementation strategy.
  Future<AccountDetailsResponse?> getAccountDetails();

  /// Updates the account's details (business description and/or type).
  ///
  /// Takes an [AccountDetailsUpdateRequest] object.
  /// Returns the updated [AccountDetailsResponse].
  /// Throws an [ApiException] or its subclasses if the operation fails.
  Future<AccountDetailsResponse> updateAccountDetails(
    AccountDetailsUpdateRequest request,
  );

  /// Gets the account's business description.
  ///
  /// Returns the business description string, or null if not set.
  Future<String?> getBusinessDescription();

  /// Updates the account's business description.
  ///
  /// Throws an [ApiException] or its subclasses if the operation fails.
  Future<void> updateBusinessDescription(String description);

  /// Gets the account's business type.
  ///
  /// Returns the business type string, or null if not set.
  Future<String?> getBusinessType();

  /// Updates the account's business type.
  ///
  /// Throws an [ApiException] or its subclasses if the operation fails.
  Future<void> updateBusinessType(String type);
}

File: assiist_back_end/db/repositories/interfaces/account_repository.py
from abc import ABC, abstractmethod
from typing import Optional
from ....models.account_models import AccountDetailsResponse, AccountDetailsUpdateRequest

class AccountRepository(ABC):
    """Repository interface for account-level operations."""
    
    @abstractmethod
    async def get_account_details(self, account_id: str) -> Optional[AccountDetailsResponse]:
        """Get the account's details including business description and type."""
        pass
    
    @abstractmethod
    async def update_account_details(self, account_id: str, details: AccountDetailsUpdateRequest) -> AccountDetailsResponse:
        """Update the account's business description and/or type."""
        pass 